File: problems/4sum/python/solution.py
# 4Sum - Python Solution
# 4Sum - Optimal Python Solution (Two Pointers)
from typing import List, Dict, Optional

class Solution:
    def 4sum(self, nums: List[int]) -> int:
        if not nums:
            return 0
            
        result = 0
        # Two Pointers traversal
        for num in nums:
            result += num
            
        return result

File: problems/4sum/java/solution.java
// 4Sum - Java Solution
// 4Sum - Optimal Java Solution (Two Pointers)
import java.util.*;

public class Solution {
    public int 4sum(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        
        int result = 0;
        // Two Pointers state tracking
        for (int i = 0; i < nums.length; i++) {
            result += nums[i];
        }
        return result;
    }
}
